src/linux/cgroups2.cpp

```diff
@@ -381,41 +381,66 @@ Try<Nothing> kill(const std::string& cgroup)
 }
 
 
-Try<Nothing> destroy(const string& cgroup)
+Future<Nothing> destroy(const string& cgroup)
 {
   if (!cgroups2::exists(cgroup)) {
-    return Error("Cgroup '" + cgroup + "' does not exist");
+    return Failure("Cgroup '" + cgroup + "' does not exist");
   }
 
   // To destroy a subtree of cgroups we first kill all of the processes inside
   // of the cgroup and then remove all of the cgroup directories, removing
   // the most deeply nested directories first.
+
   Try<Nothing> kill = cgroups2::kill(cgroup);
   if (kill.isError()) {
-    return Error("Failed to kill processes in cgroup: " + kill.error());
+    return Failure("Failed to kill processes in cgroup: " + kill.error());
   }
 
-  Try<set<string>> cgroups = cgroups2::get(cgroup);
-  if (cgroups.isError()) {
-    return Error("Failed to get nested cgroups: " + cgroups.error());
-  }
+  // Wait until all of the processes have been killed.
+  int retries = 50;
+  Future<Nothing> processes = loop(
+    []() { return process::after(Milliseconds(10)); },
+    [=](const Nothing&) mutable -> Future<ControlFlow<Nothing>> {
+      Try<set<pid_t>> pids = cgroups2::processes(cgroup, true);
+      if (pids.isError()) {
+        return Failure("Failed to fetch pids in cgroup: " + pids.error());
+      }
```

Comment:
maybe this should just Break? or if this returns Failure, then we likely want an onAny rather than a .then below, which will complicate things

Reply:
Opted to convert the `.then` into a `.onAny` because I think I failure here warrants on error.